File: Problem1/multiples.py
def findSum(initial, final, multiples):
  
  if isinstance(multiples, list) and len(multiples) > 1:
    multiplesString = "%s and %s" % (", ".join([str(i) for i in multiples[:-1]]), multiples[len(multiples)-1])
  elif isinstance(multiples, list) and len(multiples) == 1:
    multiplesString = "%s" % (multiples[0])
  else:
    multiplesString = multiples

  if initial < final:
    smaller, bigger = initial, final

  elif initial > final: 
    smaller, bigger = final, initial
  
  else:
    return "Please insert an interval between 2 different numbers"

  result = 0

  if isinstance(multiples, list):
    for x in range(round(smaller), round(bigger)):
      checkIfSame = None
      for y in multiples:
        if x % y == 0 and not x == checkIfSame:
          result += x
          checkIfSame = x
  
  else:
    for x in range(round(smaller), round(bigger)):
      if x % multiples == 0:
        result += x

  if result > 0:
    return "The sum of all the multiples of %s between %s and %s is %s" % (multiplesString, initial, final, result) 
  else: 
    return "There are no multiples of %s between %s and %s" % (multiplesString, initial, final)

print(findSum(0, 1000, [3,5]))

File: Problem1/multiples.js
function findAnySum(initial, final, multiplesOf) {

    let multiplesOfArray = JSON.parse(`[${multiplesOf}]`)
  
    let textMultiples = multiplesOfArray.length > 1 ? `${multiplesOfArray.slice(0, (multiplesOfArray.length - 1)).join(', ')} e ${multiplesOfArray[multiplesOfArray.length - 1]}` : multiplesOfArray.join('')
  
    let result = 0
  
    if (initial < final) {
      var smaller = initial
      var bigger = final
    } else if (initial > final) {
      var smaller = final
      var bigger = initial
    } else {
        return 'Please insert an interval between 2 different numbers'
    }
  
    for (let current = Math.round(smaller); current < Math.round(bigger); current++) {
      let checkIfSame
      multiplesOfArray.forEach(multiple => {
        if (current % multiple === 0 && current != checkIfSame) {
          result += current;
          checkIfSame = current
        }
      })
    }
    if (result === 0) {
      return `There are no multiples of ${textMultiples} between ${initial} and ${final}`
    } else {
      return `Solution result: The sum of all multiples of ${textMultiples} between ${initial} and ${final} is ${result}`
    }
  }

  console.log(findAnySum(0, 1000, [3, 5]));